src/Illuminate/Foundation/Testing/Constraints/HasDatabaseCount.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Illuminate\Foundation\Testing\Constraints;
+
+use Illuminate\Database\Connection;
+use PHPUnit\Framework\Constraint\Constraint;
+
+class HasDatabaseCount extends Constraint
+{
+    /**
+     * The table to test against.
+     *
+     * @var string
+     */
+    protected $table;
+
+    /**
+     * The database connection.
+     *
+     * @var \Illuminate\Database\Connection
+     */
+    protected $database;
+
+    /**
+     * The data that will be used to narrow the search in the database table.
+     *
+     * @var array
+     */
+    protected $data;
+
+    /**
+     * Create a new constraint instance.
+     *
+     * @param  string  $table
+     * @param  \Illuminate\Database\Connection  $database
+     * @param  array  $data
+     * @return void
+     */
+    public function __construct($table, Connection $database, array $data)
+    {
+        parent::__construct();
+
+        $this->table = $table;
+
+        $this->data = $data;
+
+        $this->database = $database;
+    }
+
+    /**
+     * Check if the data is found in the given table.
+     *
+     * @param  int  $count
+     * @return bool
+     */
+    public function matches($count)
+    {
+        return $this->database->table($this->table)->where($this->data)->count() == $count;
+    }
+
+    /**
+     * Get the description of the failure.
+     *
+     * @param  int  $count
+     * @return string
+     */
+    public function failureDescription($count)
+    {
+        return sprintf(
+            'count of %d rows in table [%s] matches the attributes %s',
```

Comment:
This message could be `count of x row in table y is z`

Reply:
`Expected [$count] rows in table [$table], found [$x]`

Reply:
👎 

Why? I bet `[$table]` is only done because the existing constraints do it, but they don't "special quote" (?) other dynamic values.

I wouldn't even know why especially the table needs to be in `[``]` but since existing code does it, consistency is good; doesn't mean this has to be carried forward especially there's no prior code doing that.

TL;DR: my sentiments would be to leave the PR as it is, it's most consistent with existing code 👍